### ISSUE TITLE
read timeout does not work well in Windows

### ISSUE THREAD

**Issue by wonderful-panda** [Shougo/vimproc.vim#261]
In Windows, when no data to be read, read() waits at most 16 times longer than specified timeout.

I tested this behaviour by below function.

```vim
function! VimProcTimeoutTest(timeout)
  let p = vimproc#popen3("cmd")
  call p.stdin.write("echo foo\n")
  let start = reltime()
  let err = p.stderr.read(-1, a:timeout)
  let elapsed = reltimestr(reltime(start))
  return elapsed
endfunction
```

Results of `VimProcTimeoutTest(100)`  were in the 0.16s to 1.5s range.

The reason of this behavior is that system timer resolution of Windows is 16ms by default.
So, `Sleep(1)` takes 16ms at worst, but [this code](https://github.com/Shougo/vimproc.vim/blob/b2255c66a3dc04fba1adbda3e380facff45fe6ec/src/proc_w32.c#L695) expects `Sleep(1)` takes just 1ms:

**Comment by wonderful-panda** [comment]
I wrote patch for this problem (#260).

Existing tests are passed locally, but failed in AppVeyor with mingw32 build.
I don't know what is wrong yet.

**Comment by k-takata** [comment]
> I don't know what is wrong yet.

I don't know either, but LGTM.

**Comment by Shougo** [comment]
It is fixed in #260
